### PR TITLE
fix(shell-init): handle empty path and DRY syntax tests

### PR DESCRIPTION
**Pull request by sadiksaifi**
## Summary
- Treat empty `--print-path` output as explicit failure (error message + return 1) instead of silently succeeding without `cd`
- Use `cd --` to safely handle paths starting with a hyphen
- Extract `assert_valid_shell_syntax` helper to DRY the three shell syntax tests

Addresses review feedback from #88 ([review](https://github.com/sadiksaifi/trench/pull/88#pullrequestreview-3878895584)).

## Test plan
- [x] `cargo test commands::shell_init` — all 17 tests pass
- [x] `cargo clippy` — no errors

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling in shell initialization with explicit error messaging when path operations fail or return empty results.

* **Tests**
  * Expanded test coverage for edge cases in shell initialization and added validation tests across supported shells.

### CODE REVIEW

File: src/cli/commands/shell_init.rs [hunk 1]
```diff
@@ -160,70 +166,55 @@ mod tests {
...
+    }
+
+    #[test]
+    fn fish_output_reports_error_on_empty_path() {
+        let output = generate(ShellType::Fish);
+        assert!(
+            output.contains("switch returned empty path"),
+            "fish output should report error when switch returns empty path"
+        );
+    }
```
[the remark this answers is not in the document]
Comment (sadiksaifi) [reply src/cli/commands/shell_init.rs:184]: @coderabbitai Good point — tightened both tests to also assert the `return 1` follows the error message on stderr. Fixed in 527642c.
